app/services/paymentsAndCharges/PaymentsAndChargesService.scala

```diff
@@ -98,10 +98,12 @@ class PaymentsAndChargesService @Inject()() {
       .toSeq
       .sortWith(_._1 < _._1)
 
-  def getUpcomingCharges(schemeFS: Seq[SchemeFS]): Seq[SchemeFS] =
-    schemeFS
-      .filter(_.dueDate.nonEmpty)
-      .filter(_.dueDate.get.isAfter(DateHelper.today))
+  def extractUpcomingCharges[A](seqOfCharges: Seq[A], dateToFilterOn: A => Option[LocalDate]): Seq[A] =
+    seqOfCharges.filter{ charge =>
+      val optionDate = dateToFilterOn(charge)
+      optionDate.nonEmpty &&
+        (optionDate.get.isEqual(DateHelper.today) || optionDate.get.isAfter(DateHelper.today))
+    }
```

Comment:
Did this function need to be completely refactored and renamed? Does it need to be generic? Could just adding `.filter(_.dueDate.get.isEqual(DateHelper.today))` have done the job?

Reply:
It's used by both payments and penalties so I couldn't see a way other than duplicating the method to allow both of these to use the same method.

Reply:
Ah yes, I see it being used for PsaFS. So it does need to be generic. Is rest of the refactor necessary?

Reply:
Actually I see without a defined type at the point of filtering, you won't be able to filter as it is currently

Reply:
Well that's a moot point. I did consider that when I was doing it. We could just duplicate the method, i.e. one for schemes one for penalties, but then we have code duplication.